test/server/routes/postUser.js

```diff
@@ -61,10 +61,11 @@ test('testing /api/v1/users/new, case3: validation', (t) => {
   supertest(app)
     .post('/api/v1/users/new')
     .send(user)
-    .expect(500)
-    .expect('content-type', /html/)
-    .end((err) => {
+    .expect(422)
+    .expect('content-type', /json/)
+    .end((err, response) => {
       if (err) t.error(err);
+      else t.equal(response.body.message, 'Validation Error', 'validation error');
```

Comment:
@IsraaSulaiman I know this PR is already merged, but as a suggestion for the future: If possible, try to avoid if-statements in your tests.  Ideally, you can make the conditions in your test predictable enough so that you do not need if-statements, because the test input and output is already the same.  Having tests that don't need if-statements makes it easier for the next person to figure out what's happening if they make a code change that causes the test to start failing on their machine.

Reply:
@tomduggan85  Thanks for taking the time to review our project
please feel free to comment or ask for changes any time

Just to make sure that I understood you correctly

it should be like this: 
```js
.end(err, response) => {
  t.error(err);
  t.equal(response.body.message, 'Validation Error', 'validation error');
  t.end();
}
```
This should check each time that `err` is `falsy` otherwise, it'll print the err.message
but when she used `if else` it will not check if `err` is `false` but it'll print the error if an error happen?

Reply:
@ashatat Hi! Sorry for being vague - I can't tell if this test will always have `err` truthy, or always falsy, but I think my idea was that the test should always run in the same way - either `err` will always be truthy in this test, or it'll always be falsy, so you shouldn't need an if-statement to handle two possible outcomes - your test should only expect one outcome, every time.

Reply:
@tomduggan85  Thanks, I get it now :+1: